consensus/propagation/have_wants.go

```diff
@@ -185,13 +181,16 @@ func (blockProp *Reactor) handleWants(peer p2p.ID, wants *proptypes.WantParts) {
 	// the peer must always send the proposal before sending parts, if they did
 	//  not, this node must disconnect from them.
 	if !has {
+		blockProp.Logger.Error("received part state request for unknown proposal", "peer", peer, "height", height, "round", round)
+		blockProp.Switch.StopPeerForError(p.peer, errors.New("received want part for unknown proposal"))
 		return
 	}
```

Comment:
same comment as elsewhere, pruning makes this really difficult

Reply:
05b0056448553ddddedf781aaf34074e3347f21d